File: Exercises/OpenMP/Prime/prime_serial.c
# include <stdlib.h>
# include <stdio.h>

%% #pragma omp parallel 
int main ( int argc, char *argv[] ){
  int n = 50000;
  int not_primes=0;
  int i,j;

%% #pragma omp for reduction(+:not_primes)
  for ( i = 2; i <= n; i++ ){
    for ( j = 2; j < i; j++ ){
      if ( i % j == 0 ){
	not_primes++;
	break;
      }
    }
  }

  printf("Primes: %d\n", n - not_primes);

}

File: Exercises/OpenMP/Prime/prime_serial.f
      program primes

      integer n, not_primes, i, j

      n = 500000
      not_primes=0

      do i = 2,n
         do j = 2,i-1
            if (mod(i,j) == 0) then
               not_primes = not_primes + 1
               exit
            end if
         end do
      end do

      print *, 'Primes: ', n - not_primes

      end program

